src/WallToolPaths.h

```diff
@@ -83,6 +84,30 @@ class WallToolPaths
      */
     static bool removeEmptyToolPaths(VariableWidthPaths& toolpaths);
 
+    /*!
+     * Get the order constraints of the insets assuming the Wall Ordering is outer to inner.
+     * Each returned pair consists of adjacent wall lines where the left has an inset_idx one lower than the right.
+     * 
+     * Odd walls should always go after their enclosing wall polygons.
+     * 
+     * \param outer_to_inner Whether the wall polygons with a lower inset_idx should go before those with a higher one.
+     */
```

Comment:
A weak order is mathematically defined as that the order between two elements is not given for each combination of two elements. All elements in the natural numbers are ordered, so it's called a strong order. Not all elements in the toolpaths are ordered, because no order requirement is given between two walls of the same inset.

The function you suggest is already implemented in that branch. You are looking at outdated code.

Reply:
Is a weak ordering the same as a partially ordered set? https://en.wikipedia.org/wiki/Partially_ordered_set